### ISSUE TITLE
Slow

### ISSUE THREAD

**Issue by jacobg** [Donaldcwl/browser-image-compression#215]
This library has a very nice specific purpose with a convenient function signature.

However, some feedback in comparing its speed to blueimp-load-image is that browser-image-compression is much slower, e.g., compressing a 6 MB png to 2 MB takes less than a second with blueimp-load-image, whereas it takes around 10 seconds with browser-image-compression.

For reference here is the code in blueimp-load-image to compress to a specific size:
```ts
  const image = await loadImage(file, { orientation: true })

  // We reduce by an extra 10%, because for some reason the theoretical compression factor
  // is still a little too high. Maybe there's overhead?
  const compressFactor = Math.sqrt(maxFileSize / file.size) * .9
  const canvas = await loadImage.scale(image.image, {
    maxWidth: image.originalWidth! * compressFactor,
    maxHeight: image.originalHeight! * compressFactor,
    canvas: true })

  return await new Promise<File>(resolve => {
    canvas.toBlob(
      blob => resolve(new File([blob!], file.name, { type: file.type })),
      file.type
    )
  })
```

**Comment by qkreltms** [comment]
I'm wondering have you tried `useWebWorker: true` ?

**Comment by uberubert** [comment]
I noticed a 2.6MB png took ca 5 seconds to load. I tried setting `useWebWorker: false` and `useWebWorker: true`. Both cases resulted in the same ~5 second timespan. The only real difference was how the browser was frozen with `useWebWorker: false`

Measured using `console.time('loadImage')` before running loadImage() and `console.timeEnd('loadImage')` after the result is ready.

**Comment by IndieHobDev** [comment]
It turns out that you need to use both options maxSizeMB and maxWidthOrHeight for the compression to work super fast. 11 MB images were taking close to 8-12 seconds for me, but since adding both options the compression per image takes less than 1 second.
Try it :-)

**Comment by zg19990220** [comment]
On some mobile devices, a 600kb image takes more than 30s

**Comment by Ibrahimsyah** [comment]
> It turns out that you need to use both options maxSizeMB and maxWidthOrHeight for the compression to work super fast. 11 MB images were taking close to 8-12 seconds for me, but since adding both options the compression per image takes less than 1 second. Try it :-)

yeah you are right, seems maxSizeMB doesn't work well if it not combined with maxWidthOrHeight

**Comment by diamont1001** [comment]
If you don't have multiple images at the same time, you can try `useWebWorker: false`

`webworker.js` will take a long time to download sometimes.

**Comment by xiaoxuwei12345** [comment]
I usually use https://www.freeimagecompression.com
 for online image compression. It runs entirely in the browser, is very secure, and extremely fast.
